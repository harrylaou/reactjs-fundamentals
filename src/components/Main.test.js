import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ConnectedMain, {Main} from './Main'


describe('<Main />', () => {
  it('should render with default props', () => {
    const logoutUser = jest.fn();
    const props = {
      width: 3
    }

    const wrapper = shallow(
      <ConnectedMain {...props} />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.find('Main').simulate('click')
  })

  describe('Testing Methods', () => {
    let wrapper

    beforeEach(() => {
      const props = {
        width: 3
      }

      wrapper = shallow(
        <Main {...props} />
      )
      wrapper.instance().setState = jest.fn()
    })

    it('should mount and map connect connected component', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot()
    })

    it('should check #closeNav()', () => {
      wrapper.instance().toggleNav()

      expect(wrapper.instance().setState).toBeCalledWith({ nav: { open: true } })

    })

    it('should check #closeNav()', () => {
      wrapper.instance().closeNav()

      expect(wrapper.instance().setState).toBeCalledWith({ nav: { open: false } })
    })
  })
})