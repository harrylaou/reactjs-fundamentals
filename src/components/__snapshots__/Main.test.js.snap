// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Testing Methods should mount and map connect connected component 1`] = `
<MuiThemeProvider
  muiTheme={
    Object {
      "appBar": Object {
        "color": "#880e4f",
        "height": 64,
        "padding": 24,
        "textColor": "#303030",
        "titleFontWeight": 400,
      },
      "avatar": Object {
        "backgroundColor": "rgb(101, 101, 101)",
        "color": "#303030",
      },
      "badge": Object {
        "color": "#303030",
        "fontWeight": 500,
        "primaryColor": "#ff4081",
        "primaryTextColor": "#303030",
        "secondaryColor": "#880e4f",
        "secondaryTextColor": "#303030",
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "baseTheme": Object {
        "fontFamily": "Roboto, sans-serif",
        "palette": Object {
          "accent1Color": "#ff4081",
          "accent2Color": "#f50057",
          "accent3Color": "#ff80ab",
          "alternateTextColor": "#303030",
          "borderColor": "rgba(255, 255, 255, 0.3)",
          "canvasColor": "#303030",
          "clockCircleColor": "rgba(255, 255, 255, 0.12)",
          "disabledColor": "rgba(255, 255, 255, 0.3)",
          "pickerHeaderColor": "rgba(255, 255, 255, 0.12)",
          "primary1Color": "#880e4f",
          "primary2Color": "#880e4f",
          "primary3Color": "#757575",
          "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
          "shadowColor": "rgba(0, 0, 0, 1)",
          "textColor": "rgba(255, 255, 255, 1)",
        },
        "spacing": Object {
          "desktopDrawerMenuItemHeight": 48,
          "desktopDropDownMenuFontSize": 15,
          "desktopDropDownMenuItemHeight": 32,
          "desktopGutter": 24,
          "desktopGutterLess": 16,
          "desktopGutterMini": 8,
          "desktopGutterMore": 32,
          "desktopKeylineIncrement": 64,
          "desktopSubheaderHeight": 48,
          "desktopToolbarHeight": 56,
          "iconSize": 24,
        },
      },
      "bottomNavigation": Object {
        "backgroundColor": "#303030",
        "height": 56,
        "selectedColor": "#880e4f",
        "selectedFontSize": 14,
        "unselectedColor": "rgba(255, 255, 255, 0.54)",
        "unselectedFontSize": 12,
      },
      "button": Object {
        "height": 36,
        "iconButtonSize": 48,
        "minWidth": 88,
      },
      "card": Object {
        "fontWeight": 500,
        "subtitleColor": "rgba(255, 255, 255, 0.54)",
        "titleColor": "rgba(255, 255, 255, 0.87)",
      },
      "cardMedia": Object {
        "color": "rgba(255, 255, 255, 0.87)",
        "overlayContentBackground": "rgba(0, 0, 0, 0.54)",
        "subtitleColor": "rgba(255, 255, 255, 0.54)",
        "titleColor": "rgba(255, 255, 255, 0.87)",
      },
      "cardText": Object {
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "checkbox": Object {
        "boxColor": "rgba(255, 255, 255, 1)",
        "checkedColor": "#880e4f",
        "disabledColor": "rgba(255, 255, 255, 0.3)",
        "labelColor": "rgba(255, 255, 255, 1)",
        "labelDisabledColor": "rgba(255, 255, 255, 0.3)",
        "requiredColor": "#880e4f",
      },
      "chip": Object {
        "backgroundColor": "rgb(72, 72, 72)",
        "deleteIconColor": "rgba(255, 255, 255, 0.26)",
        "fontSize": 14,
        "fontWeight": 400,
        "shadow": "0 1px 6px rgba(0, 0, 0, 0.12),
            0 1px 4px rgba(0, 0, 0, 0.12)",
        "textColor": "rgba(255, 255, 255, 0.87)",
      },
      "datePicker": Object {
        "calendarTextColor": "rgba(255, 255, 255, 1)",
        "calendarYearBackgroundColor": "#ffffff",
        "color": "#880e4f",
        "selectColor": "#880e4f",
        "selectTextColor": "#303030",
        "textColor": "#303030",
      },
      "dialog": Object {
        "bodyColor": "rgba(255, 255, 255, 0.6)",
        "bodyFontSize": 16,
        "titleFontSize": 22,
      },
      "drawer": Object {
        "color": "#303030",
        "width": 256,
      },
      "dropDownMenu": Object {
        "accentColor": "rgba(255, 255, 255, 0.3)",
      },
      "enhancedButton": Object {
        "tapHighlightColor": "rgba(0, 0, 0, 0)",
      },
      "flatButton": Object {
        "buttonFilterColor": "#999999",
        "color": "rgba(0, 0, 0, 0)",
        "disabledTextColor": "rgba(255, 255, 255, 0.3)",
        "fontSize": 14,
        "fontWeight": 500,
        "primaryTextColor": "#880e4f",
        "secondaryTextColor": "#ff4081",
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "floatingActionButton": Object {
        "buttonSize": 56,
        "color": "#880e4f",
        "disabledColor": "rgb(72, 72, 72)",
        "disabledTextColor": "rgba(255, 255, 255, 0.3)",
        "iconColor": "#303030",
        "miniSize": 40,
        "secondaryColor": "#ff4081",
        "secondaryIconColor": "#303030",
      },
      "fontFamily": "Roboto, sans-serif",
      "gridTile": Object {
        "textColor": "#ffffff",
      },
      "icon": Object {
        "backgroundColor": "#880e4f",
        "color": "#303030",
      },
      "inkBar": Object {
        "backgroundColor": "#ff4081",
      },
      "isRtl": false,
      "listItem": Object {
        "leftIconColor": "#757575",
        "nestedLevelDepth": 18,
        "rightIconColor": "#757575",
        "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
      },
      "menu": Object {
        "backgroundColor": "#303030",
        "containerBackgroundColor": "#303030",
      },
      "menuItem": Object {
        "dataHeight": 32,
        "height": 48,
        "hoverColor": "rgba(255, 255, 255, 0.035)",
        "padding": 24,
        "rightIconDesktopFill": "#757575",
        "selectedTextColor": "#ff4081",
      },
      "menuSubheader": Object {
        "borderColor": "rgba(255, 255, 255, 0.3)",
        "padding": 24,
        "textColor": "#880e4f",
      },
      "overlay": Object {
        "backgroundColor": "rgba(0, 0, 0, 0.54)",
      },
      "palette": Object {
        "accent1Color": "#ff4081",
        "accent2Color": "#f50057",
        "accent3Color": "#ff80ab",
        "alternateTextColor": "#303030",
        "borderColor": "rgba(255, 255, 255, 0.3)",
        "canvasColor": "#303030",
        "clockCircleColor": "rgba(255, 255, 255, 0.12)",
        "disabledColor": "rgba(255, 255, 255, 0.3)",
        "pickerHeaderColor": "rgba(255, 255, 255, 0.12)",
        "primary1Color": "#880e4f",
        "primary2Color": "#880e4f",
        "primary3Color": "#757575",
        "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
        "shadowColor": "rgba(0, 0, 0, 1)",
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "paper": Object {
        "backgroundColor": "#303030",
        "color": "rgba(255, 255, 255, 1)",
        "zDepthShadows": Array [
          "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
          "0 3px 10px rgba(0, 0, 0, 0.16),
             0 3px 10px rgba(0, 0, 0, 0.23)",
          "0 10px 30px rgba(0, 0, 0, 0.19),
             0 6px 10px rgba(0, 0, 0, 0.23)",
          "0 14px 45px rgba(0, 0, 0, 0.25),
             0 10px 18px rgba(0, 0, 0, 0.22)",
          "0 19px 60px rgba(0, 0, 0, 0.3),
             0 15px 20px rgba(0, 0, 0, 0.22)",
        ],
      },
      "prepareStyles": [Function],
      "radioButton": Object {
        "backgroundColor": "#303030",
        "borderColor": "rgba(255, 255, 255, 1)",
        "checkedColor": "#880e4f",
        "disabledColor": "rgba(255, 255, 255, 0.3)",
        "labelColor": "rgba(255, 255, 255, 1)",
        "labelDisabledColor": "rgba(255, 255, 255, 0.3)",
        "requiredColor": "#880e4f",
        "size": 24,
      },
      "raisedButton": Object {
        "color": "#303030",
        "disabledColor": "rgb(43, 43, 43)",
        "disabledTextColor": "rgba(255, 255, 255, 0.3)",
        "fontSize": 14,
        "fontWeight": 500,
        "primaryColor": "#880e4f",
        "primaryTextColor": "#303030",
        "secondaryColor": "#ff4081",
        "secondaryTextColor": "#303030",
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "rawTheme": Object {
        "fontFamily": "Roboto, sans-serif",
        "palette": Object {
          "accent1Color": "#ff4081",
          "accent2Color": "#f50057",
          "accent3Color": "#ff80ab",
          "alternateTextColor": "#303030",
          "borderColor": "rgba(255, 255, 255, 0.3)",
          "canvasColor": "#303030",
          "clockCircleColor": "rgba(255, 255, 255, 0.12)",
          "disabledColor": "rgba(255, 255, 255, 0.3)",
          "pickerHeaderColor": "rgba(255, 255, 255, 0.12)",
          "primary1Color": "#880e4f",
          "primary2Color": "#880e4f",
          "primary3Color": "#757575",
          "secondaryTextColor": "rgba(0, 0, 0, 0.54)",
          "shadowColor": "rgba(0, 0, 0, 1)",
          "textColor": "rgba(255, 255, 255, 1)",
        },
        "spacing": Object {
          "desktopDrawerMenuItemHeight": 48,
          "desktopDropDownMenuFontSize": 15,
          "desktopDropDownMenuItemHeight": 32,
          "desktopGutter": 24,
          "desktopGutterLess": 16,
          "desktopGutterMini": 8,
          "desktopGutterMore": 32,
          "desktopKeylineIncrement": 64,
          "desktopSubheaderHeight": 48,
          "desktopToolbarHeight": 56,
          "iconSize": 24,
        },
      },
      "refreshIndicator": Object {
        "loadingStrokeColor": "#880e4f",
        "strokeColor": "rgba(255, 255, 255, 0.3)",
      },
      "ripple": Object {
        "color": "rgba(255, 255, 255, 0.87)",
      },
      "slider": Object {
        "handleColorZero": "#757575",
        "handleFillColor": "#303030",
        "handleSize": 12,
        "handleSizeActive": 18,
        "handleSizeDisabled": 8,
        "rippleColor": "#880e4f",
        "selectionColor": "#880e4f",
        "trackColor": "#757575",
        "trackColorSelected": "#ff80ab",
        "trackSize": 2,
      },
      "snackbar": Object {
        "actionColor": "#ff4081",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "textColor": "#303030",
      },
      "spacing": Object {
        "desktopDrawerMenuItemHeight": 48,
        "desktopDropDownMenuFontSize": 15,
        "desktopDropDownMenuItemHeight": 32,
        "desktopGutter": 24,
        "desktopGutterLess": 16,
        "desktopGutterMini": 8,
        "desktopGutterMore": 32,
        "desktopKeylineIncrement": 64,
        "desktopSubheaderHeight": 48,
        "desktopToolbarHeight": 56,
        "iconSize": 24,
      },
      "stepper": Object {
        "backgroundColor": "transparent",
        "connectorLineColor": "#bdbdbd",
        "disabledTextColor": "rgba(0, 0, 0, 0.26)",
        "hoverBackgroundColor": "rgba(0, 0, 0, 0.06)",
        "hoveredIconColor": "#616161",
        "iconColor": "#880e4f",
        "inactiveIconColor": "#9e9e9e",
        "textColor": "rgba(0, 0, 0, 0.87)",
      },
      "subheader": Object {
        "color": "rgba(255, 255, 255, 0.54)",
        "fontWeight": 500,
      },
      "svgIcon": Object {
        "color": "rgba(255, 255, 255, 1)",
      },
      "table": Object {
        "backgroundColor": "#303030",
      },
      "tableFooter": Object {
        "borderColor": "rgba(255, 255, 255, 0.3)",
        "textColor": "#ff80ab",
      },
      "tableHeader": Object {
        "borderColor": "rgba(255, 255, 255, 0.3)",
      },
      "tableHeaderColumn": Object {
        "height": 56,
        "spacing": 24,
        "textColor": "#ff80ab",
      },
      "tableRow": Object {
        "borderColor": "rgba(255, 255, 255, 0.3)",
        "height": 48,
        "hoverColor": "#f50057",
        "selectedColor": "rgba(255, 255, 255, 0.3)",
        "stripeColor": "rgba(195, 134, 167, 0.4)",
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "tableRowColumn": Object {
        "height": 48,
        "spacing": 24,
      },
      "tabs": Object {
        "backgroundColor": "#880e4f",
        "selectedTextColor": "#303030",
        "textColor": "rgba(48, 48, 48, 0.7)",
      },
      "textField": Object {
        "backgroundColor": "transparent",
        "borderColor": "rgba(255, 255, 255, 0.3)",
        "disabledTextColor": "rgba(255, 255, 255, 0.3)",
        "errorColor": "#f44336",
        "floatingLabelColor": "rgba(255, 255, 255, 1)",
        "focusColor": "#880e4f",
        "hintColor": "rgba(255, 255, 255, 0.3)",
        "textColor": "rgba(255, 255, 255, 1)",
      },
      "timePicker": Object {
        "accentColor": "#880e4f",
        "clockCircleColor": "rgba(255, 255, 255, 0.12)",
        "clockColor": "rgba(255, 255, 255, 1)",
        "color": "#303030",
        "headerColor": "rgba(255, 255, 255, 0.12)",
        "selectColor": "#880e4f",
        "selectTextColor": "#303030",
        "textColor": "#ff80ab",
      },
      "toggle": Object {
        "labelColor": "rgba(255, 255, 255, 1)",
        "labelDisabledColor": "rgba(255, 255, 255, 0.3)",
        "thumbDisabledColor": "rgba(255, 255, 255, 0.3)",
        "thumbOffColor": "#f50057",
        "thumbOnColor": "#880e4f",
        "thumbRequiredColor": "#880e4f",
        "trackDisabledColor": "#757575",
        "trackOffColor": "#757575",
        "trackOnColor": "rgba(136, 14, 79, 0.5)",
        "trackRequiredColor": "rgba(136, 14, 79, 0.5)",
      },
      "toolbar": Object {
        "backgroundColor": "rgb(232, 0, 82)",
        "color": "rgba(255, 255, 255, 0.54)",
        "height": 56,
        "hoverColor": "rgba(255, 255, 255, 0.87)",
        "iconColor": "rgba(255, 255, 255, 0.4)",
        "menuHoverColor": "rgba(255, 255, 255, 0.1)",
        "separatorColor": "rgba(255, 255, 255, 0.175)",
        "titleFontSize": 20,
      },
      "tooltip": Object {
        "color": "#ffffff",
        "rippleBackgroundColor": "#616161",
      },
      "userAgent": undefined,
      "zIndex": Object {
        "appBar": 1100,
        "dialog": 1500,
        "dialogOverlay": 1400,
        "drawer": 1300,
        "drawerOverlay": 1200,
        "layer": 2000,
        "menu": 1000,
        "popover": 2100,
        "snackbar": 2900,
        "tooltip": 3000,
      },
    }
  }
>
  <div
    className="main-view"
  >
    <AppBar
      className="app-bar"
      iconElementRight={
        <withRouter(Connect(LogoutContainer))
          color="rgba(255, 255, 255, 1)"
        />
      }
      onLeftIconButtonTouchTap={[Function]}
      showMenuIconButton={true}
      title="Course Manager"
      zDepth={1}
    />
    <withRouter(AppNavDrawer)
      closeNav={[Function]}
      docked={true}
      open={true}
      styles={
        Object {
          "appBar": Object {
            "color": "rgba(255, 255, 255, 1)",
            "position": "fixed",
            "top": 0,
            "zIndex": 1101,
          },
          "logo": Object {
            "backgroundColor": "#c2185b",
            "color": "rgba(255, 255, 255, 1)",
            "cursor": "pointer",
            "fontSize": 24,
            "fontWeight": 300,
            "lineHeight": "64px",
            "marginBottom": 8,
            "paddingLeft": 24,
          },
          "nav": Object {
            "color": "rgba(255, 255, 255, 1)",
          },
          "root": Object {
            "muiPrepared": true,
            "paddingLeft": 256,
          },
        }
      }
      toggleNav={[Function]}
    />
    <div
      className="view-container"
      style={
        Object {
          "muiPrepared": true,
          "paddingLeft": 256,
        }
      }
    />
  </div>
</MuiThemeProvider>
`;

exports[`<Main /> should render with default props 1`] = `
<Main
  width={3}
/>
`;
